subprojects/dependency-management/src/main/java/org/gradle/internal/locking/DependencyLockingArtifactVisitor.java

```diff
@@ -45,6 +47,7 @@
     private final String configurationName;
     private Set<String> lockingConstraints = Collections.emptySet();
     private Set<ModuleComponentIdentifier> allResolvedModules;
+    private Set<ModuleComponentIdentifier> changingResolvedModules = new HashSet<ModuleComponentIdentifier>();
```

Comment:
:x: Either you should use a `LinkedHashSet` here, or sort when you print the details in the error message, or the tests are going to be flaky if there are more than one modules. I'd also consider creating this set lazily, as in most cases you wouldn't have any changing module. This has proved to save a lot of memory allocation in the past.

Reply:
Sorting happens when we transform the identifiers to string, so that's taken care of. Will do the lazy init as indeed this should be exceptional.

Reply:
That's not true. `HashSet` doesn't have any iteration order guarantee. It's been the cause of a lot of flakiness.

Reply:
I do an explicit sort, but only when this information needs to be outputted, and based on the resulting `String`.